Userland/Libraries/LibWeb/HTML/Navigable.cpp

```diff
@@ -855,7 +855,7 @@ static WebIDL::ExceptionOr<Navigable::NavigationParamsVariant> create_navigation
         }
 
         // 7. Wait until either response is non-null, or navigable's ongoing navigation changes to no longer equal navigationId.
-        Platform::EventLoopPlugin::the().spin_until([&]() {
+        HTML::main_thread_event_loop().spin_until([&]() {
```

Comment:
I guess my question here is how do we make sure we don't pick up another task from the navigation task source while doing this spin? Iirc @kalenikaliaksandr had issues with that in the past and invented a special helper. Is this a concern for this method?

Reply:
Ah, looked at the rest of the context. Yes, this code will handle that situation gracefully. Nice work!